pkg/pulumiyaml/run.go

```diff
@@ -1578,7 +1578,7 @@ func (e *programEvaluator) evaluatePropertyAccessTail(expr ast.Expr, receiver in
 				if x.ContainsUnknowns() && !ok {
 					return unknownOutput(), true
 				} else if !ok {
-					receiver = nil
+					receiver = unknownOutput()
```

Comment:
I'm not sure a missing key should be treated the same as a computed key.
The engine _does_ send back markers for computed fields, they're not just skipped from the output.
Making everything computed here means accessing typo'd fields won't fail at preview, and will try to send computed values on update.